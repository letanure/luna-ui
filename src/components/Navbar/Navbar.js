import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import hex2Rgba from '../../ultils/hex2Rgba';

/**
 * Navbar component.
 * E.g.:
 * ```html
 * <Navbar fixed />
 * ```
 */
class NavbarComponent extends React.PureComponent {
  render() {
    const { fixed, className } = this.props;

    const Navbar = styled.div`
      width: 100%;
      height: 42px;
      z-index: 1000;
      ${fixed ? `position: fixed;` : ''}
      transition: background-color .15s linear;
      border-bottom: 1px solid ${(props) => props.theme.navbar.border};
      background-color: ${(props) => hex2Rgba(props.theme.navbar.bg, 0.8)};
      &:hover {
        background-color: ${(props) => hex2Rgba(props.theme.navbar.bg, 1)};
      }
    `;

    return (
      <Navbar className={className}>
        {this.props.children}
      </Navbar>
    );
  }
}

NavbarComponent.propTypes = {
  children: PropTypes.node,
  /**
   * set navbar position to fixed.
   */
  fixed: PropTypes.bool,
  className: PropTypes.string
};

NavbarComponent.defaultProps = {
  children: null,
  fixed: false,
  className: ''
};

export default NavbarComponent;
