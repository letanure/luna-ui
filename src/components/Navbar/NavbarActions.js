import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * NavbarActions component.
 * E.g.:
 * ```html
 * <Navbar.Actions />
 * ```
 */
class NavbarActionsComponent extends React.PureComponent {
  render() {
    const { className } = this.props;

    const NavbarActions = styled.div`
      float: right;
    `;

    return <NavbarActions className={className}>{this.props.children}</NavbarActions>;
  }
}

NavbarActionsComponent.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

NavbarActionsComponent.defaultProps = {
  children: null,
  className: ''
};

export default NavbarActionsComponent;
