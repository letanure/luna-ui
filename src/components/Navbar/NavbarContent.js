import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * NavbarContent component.
 * E.g.:
 * ```html
 * <Navbar.Content />
 * ```
 */
class NavbarContentComponent extends React.PureComponent {
  render() {
    const { className } = this.props;

    const NavbarContent = styled.div`
      display: inline-block;
    `;

    return <NavbarContent className={className}>{this.props.children}</NavbarContent>;
  }
}

NavbarContentComponent.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

NavbarContentComponent.defaultProps = {
  children: null,
  className: ''
};

export default NavbarContentComponent;
