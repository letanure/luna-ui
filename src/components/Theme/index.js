import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

import defaultTheme from '../../config/defaultTheme';
import DefaultStyles from '../../config/defaultStyles';
import DefaulTyphography from '../../config/defaultTyphography';

/**
 * Theme component.
 * E.g.:
 * ```html
 * <Theme theme={{...}}>
 *  ...
 * </Theme>
 * ```
 */
class ThemeComponent extends React.PureComponent {
  render() {
    const { theme, children } = this.props;

    return (
      <ThemeProvider theme={theme}>
        <div>
          <DefaultStyles />
          <DefaulTyphography />
          {children}
        </div>
      </ThemeProvider>
    );
  }
}

ThemeComponent.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node
};

ThemeComponent.defaultProps = {
  theme: defaultTheme,
  children: null
};

export default ThemeComponent;
