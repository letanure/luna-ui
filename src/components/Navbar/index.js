import Navbar from './Navbar';
import NavbarActions from './NavbarActions';
import NavbarContent from './NavbarContent';

Navbar.Actions = NavbarActions;
Navbar.Content = NavbarContent;

export default Navbar;
