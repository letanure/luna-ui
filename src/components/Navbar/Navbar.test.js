import React from 'react';
import renderer from 'react-test-renderer';

import Theme from '../../config/defaultTheme';
import ThemeProvider from '../Theme/index';

import Navbar from './index';

describe('<Navbar />', () => {
  const navBarContent =
    <div>
      <Navbar.Content>
        <h1>My App</h1>
      </Navbar.Content>
      <Navbar.Actions>
        <button>actions</button>
      </Navbar.Actions>
    </div>;

  const component =
    <ThemeProvider theme={Theme}>
      <Navbar>{navBarContent}</Navbar>
    </ThemeProvider>;

  const componentFixed =
    <ThemeProvider theme={Theme}>
      <Navbar fixed>{navBarContent}</Navbar>
    </ThemeProvider>;

  test('toMatchSnapshot: Static', () => {
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('toMatchSnapshot: Fixed', () => {
    const tree = renderer
      .create(componentFixed)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
