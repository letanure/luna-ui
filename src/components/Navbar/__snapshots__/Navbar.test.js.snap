// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> toMatchSnapshot: Fixed 1`] = `
<div>
  <div
    className="sc-bxivhb izemKD"
  >
    <div>
      <div
        className="sc-ifAKCX gJdvJe"
      >
        <h1>
          My App
        </h1>
      </div>
      <div
        className="sc-EHOje dNeOWL"
      >
        <button>
          actions
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Navbar /> toMatchSnapshot: Static 1`] = `
<div>
  <div
    className="sc-bdVaJa cPehbA"
  >
    <div>
      <div
        className="sc-bwzfXH bdSfMS"
      >
        <h1>
          My App
        </h1>
      </div>
      <div
        className="sc-htpNat iCvOKN"
      >
        <button>
          actions
        </button>
      </div>
    </div>
  </div>
</div>
`;
